src/benchmark/operators/table_scan_sorted_benchmark.cpp

```diff
@@ -0,0 +1,277 @@
+#include "storage/segment_encoding_utils.hpp"
+#include "storage/table.hpp"
+#include "table_generator.hpp"
+#include "type_cast.hpp"
+#include "utils/load_table.hpp"
+
+using namespace opossum::expression_functional;  // NOLINT
+
+namespace opossum {
+
+const int string_size = 512;
+
+void _clear_cache() {
+  std::vector<int> clear = std::vector<int>();
+  clear.resize(500 * 1000 * 1000, 42);
+  for (uint i = 0; i < clear.size(); i++) {
+    clear[i] += 1;
+  }
+  clear.resize(0);
+}
+
+std::shared_ptr<TableColumnDefinitions> create_column_definitions(const DataType data_type) {
+  auto table_column_definitions = std::make_shared<TableColumnDefinitions>();
+
+  // TODO(cmfcmf): Benchmark nullable segments
+  table_column_definitions->emplace_back("a", data_type, false);
+
+  return table_column_definitions;
+}
+
+std::shared_ptr<TableWrapper> create_int_table(const int table_size,
+                                               const EncodingType encoding_type = EncodingType::Unencoded,
+                                               const std::optional<OrderByMode> order_by = std::nullopt) {
+  const auto table_column_definitions = create_column_definitions(DataType::Int);
+
+  std::shared_ptr<Table> table;
+  std::shared_ptr<TableWrapper> table_wrapper;
+
+  table = std::make_shared<Table>(*table_column_definitions, TableType::Data);
+
+  if (!order_by.has_value()) {
+    auto values = std::vector<int>(table_size);
+    for (int i = 0; i < table_size; i++) {
+      values[i] = i;
+    }
+    std::random_shuffle(values.begin(), values.end());
+    for (const auto& num : values) {
+      table->append({num});
+    }
+  } else {
+    for (int i = 0; i < table_size; i++) {
+      if (order_by.value() == OrderByMode::Ascending || order_by.value() == OrderByMode::AscendingNullsLast) {
+        table->append({i});
+      } else {
+        table->append({table_size - i - 1});
+      }
+    }
+  }
+
+  if (encoding_type != EncodingType::Unencoded) {
+    ChunkEncoder::encode_all_chunks(table, SegmentEncodingSpec(encoding_type));
+  }
+
+  if (order_by.has_value()) {
+    for (auto& chunk : table->chunks()) {
+      chunk->get_segment(ColumnID(0))->set_sort_order(order_by.value());
+    }
+  }
+
+  table_wrapper = std::make_shared<TableWrapper>(std::move(table));
+  table_wrapper->execute();
+
+  return table_wrapper;
+}
+
+std::shared_ptr<TableWrapper> create_string_table(const int table_size, const int string_length,
+                                                  const EncodingType encoding_type = EncodingType::Unencoded,
+                                                  const std::optional<OrderByMode> order_by = std::nullopt) {
+  const auto table_column_definitions = create_column_definitions(DataType::String);
+  std::shared_ptr<Table> table;
+  std::shared_ptr<TableWrapper> table_wrapper;
+
+  table = std::make_shared<Table>(*table_column_definitions, TableType::Data);
+
+  if (!order_by.has_value()) {
```

Comment:
I suggest the alternative implementation from above to reduce code duplication. In general, one could further argue, that `create_int_table` and `create_string_table` are pretty much identical. I don't know, however, if it is worth the effort to refactor these two functions to prevent code duplication, since we are just dealing with a simple benchmark.